django/utils/http.py

```diff
@@ -290,6 +290,8 @@ def is_safe_url(url, host=None):
         url = url.strip()
     if not url:
         return False
+    if six.PY2:
+        url = force_text(url, errors='replace')
```

Comment:
I'm wondering about `errors='replace'` here.  The function below considers control characters.  Is it possible that some of the bad characters we're trying to detect here will be turned into replacement characters, and then overlooked by `_is_safe_url` below?  The Unicode replacement character is U+FFFD, which is category "So".